sentry-types/src/protocol/profile.rs

```diff
@@ -1,77 +1,143 @@
+    /// List of stacks: each stacks is a vec of indexed frames
+    pub stacks: Vec<Vec<u32>>,
+    /// List of frames
+    pub frames: Vec<RustFrame>,
+    /// Thread metadata
+    pub thread_metadata: HashMap<String, ThreadMetadata>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq)]
+/// Operating System metadata
+pub struct OSMetadata {
+    /// OS Name
+    pub name: String,
+    /// OS Version
+    pub version: String,
+
     #[serde(default, skip_serializing_if = "Option::is_none")]
+    /// Build number
+    pub build_number: Option<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq)]
+/// Runtime metadata
+pub struct RuntimeMetadata {
+    /// Runtime name (rustc)
+    pub name: String,
+    /// Runtime version
+    pub version: String,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq)]
+/// Device metadata
+pub struct DeviceMetadata {
+    /// Architecture
     pub architecture: Option<String>,
-    /// The trace ID
-    pub trace_id: TraceId,
-    /// The name of the transaction this profile belongs to
-    pub transaction_name: String,
-    #[serde(serialize_with = "serialize_id")]
-    /// The ID of the transaction this profile belongs to
-    pub transaction_id: Uuid,
-    /// The ID of the event
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq)]
+/// Profile format version
+pub enum Version {
+    #[serde(rename = "1")]
+    /// First version
+    V1,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
+/// Represents a Profile Envelope ItemType
+pub struct SampleProfile {
+    /// Format version of the SampleProfile
+    pub version: Version,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    /// Debug meta information
+    pub debug_meta: Option<DebugMeta>,
+
+    /// Device metadata information
+    pub device: DeviceMetadata,
+    /// OS metadata information
+    pub os: OSMetadata,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    /// Runtime metadata information
+    pub runtime: Option<RuntimeMetadata>,
+
+    #[serde(default, skip_serializing_if = "String::is_empty")]
+    /// Environment
+    pub environment: String,
     #[serde(serialize_with = "serialize_id")]
-    pub profile_id: Uuid,
-    /// Represents the profile collected
-    pub sampled_profile: SampledProfile,
-    /// OS name
-    #[serde(rename = "device_os_name")]
-    pub os_name: String,
-    #[serde(rename = "device_os_version")]
-    /// OS version
-    pub os_version: String,
-    /// Package version
-    pub version_name: String,
-    /// Current binary build ID. See <https://docs.rs/build_id/latest/build_id/>
-    pub version_code: String,
+    /// Event ID or Profile ID
+    pub event_id: Uuid,
+    /// Platform
+    pub platform: String,
+    /// Collected profile
+    pub profile: Profile,
+    /// Release
+    pub release: String,
+    /// Timestamp at which the profiler started
+    pub timestamp: DateTime<Utc>,
```

Comment:
This would be `SystemTime` then.

Reply:
Why so? 

In the backend it expects something like this `"timestamp": "2022-09-01T09:45:00.000Z"` so [this](https://docs.rs/chrono/0.4.0/chrono/struct.DateTime.html) looked like the way to go (unless we are discarding _chrono_ as discussed in the comments above)

Reply:
The relevant fields in `sentry-types` have custom serde serializers that are taking care of this, for example `#[serde(default = "SystemTime::now", with = "ts_rfc3339")]`. There is `sentry_types::utils::ts_rfc3339` that does this.